src/util/cached_chain.rs

```diff
@@ -55,8 +55,7 @@ pub fn sys() -> Option<PathBuf> {
 pub fn path() -> Vec<PathBuf> {
     vec![env_var(), home(), sys()]
         .into_iter()
-        .filter(|p| p.is_some())
-        .map(|p| p.unwrap())
+        .flatten()
```

Comment:
This should be in a separate commit because it is a logical change of its own and not related to the other change in this commit.